source/utilities.tex

```diff
@@ -11504,6 +11545,462 @@
+\begin{itemdecl}
+~out_ptr_t();
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+Let \tcode{SP} be
+\tcode{\exposid{POINTER_OF_OR}(Smart, Pointer)}\iref{memory.general}.
+
+\pnum
+\effects
+Equivalent to:
+\begin{itemize}
+\item
+% pretend to \item that there is real text here, but undo the vertical spacing
+\mbox{}\vspace{-\baselineskip}\vspace{-\parskip}
+\begin{codeblock}
+if (p) {
+  apply([&](auto&&... args) {
+    s.reset(static_cast<SP>(p), std::forward<Args>(args)... ); }, std::move(a));
+}
+\end{codeblock}
+if the expression
+\tcode{s.reset(static_cast<SP>(p), std::forward<Args>(args)...)}
+is well-\newline formed;
+\item
+otherwise,
+\begin{codeblock}
+if (p) {
+  apply([&](auto&&... args) {
+    s = Smart(static_cast<SP>(p), std::forward<Args>(args)... ); }, std::move(a));
+}
+\end{codeblock}
+if \tcode{is_constructible_v<Smart, SP, Args...>} is \tcode{true};
+\item
+otherwise, the program is ill-formed.
+\end{itemize}
+\end{itemdescr}
+
+\begin{itemdecl}
+operator Pointer*() const noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\expects
+\tcode{operator void**()} has not been called on \tcode{*this}.
+
+\pnum
+\returns
+\tcode{addressof(const_cast<Pointer\&>(p))}.
+\end{itemdescr}
+
+\begin{itemdecl}
+operator void**() const noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\constraints
+\tcode{is_same_v<Pointer, void*>} is \tcode{false}.
+
+\pnum
+\mandates
+\tcode{is_pointer_v<Pointer>} is \tcode{true}.
+
+\pnum
+\expects
+\tcode{operator Pointer*()} has not been called on \tcode{*this}.
+
+\pnum
+\returns
+A pointer value \tcode{v} such that:
+\begin{itemize}
+\item
+the initial value \tcode{*v} is equivalent to \tcode{static_cast<void*>(p)} and
+\item
+any modification of \tcode{*v}
+that is not followed by a subsequent modification of \tcode{*this}
+affects the value of \tcode{p} during the destruction of \tcode{*this},
+such that \tcode{static_cast<void*>(p) == *v}.
+\end{itemize}
+
+\pnum
+\remarks
+Accessing \tcode{*v} outside the lifetime of \tcode{*this}
+has undefined behavior.
+
+\pnum
+\begin{note}
+\tcode{reinterpret_cast<void**>(static_cast<Pointer*>(*this))}
+can be a viable implementation strategy.
+\end{note}
+\end{itemdescr}
+
+\rSec2[out.ptr]{Function template \tcode{out_ptr}}
+
+\indexlibraryglobal{out_ptr}%
+\begin{itemdecl}
+template<class Pointer = void, class Smart, class... Args>
+  auto out_ptr(Smart& s, Args&&... args);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+Let \tcode{P} be \tcode{Pointer}
+if \tcode{is_void_v<Pointer>} is \tcode{false},
+otherwise \tcode{\exposid{POINTER_OF}(Smart)}.
+
+\pnum
+\returns
+\tcode{out_ptr_t<Smart, P, Args\&\&...>(s, std::forward<Args>(args)...)}
+\end{itemdescr}
+
+\rSec2[inout.ptr.t]{Class template \tcode{inout_ptr_t}}
+
+\pnum
+\tcode{inout_ptr_t} is a class template used types
+such as smart pointers\iref{smartptr},
+with functions that use output pointer parameters
+whose dereferenced values may first be deleted
+before being set to another allocated value.
```

Comment:
Looks like something's missing between "used" and "types".

Reply:
Should be "used to adapt types" again

Reply:
```suggestion
\pnum
\tcode{inout_ptr_t} is a class template used to adapt types
such as smart pointers\iref{smartptr},
with functions that use output pointer parameters
whose dereferenced values may first be deleted
before being set to another allocated value.
```

Reply:
fixed